pkg/auth/handlers/handlers.go

```diff
@@ -60,6 +60,9 @@ func WithAuthentication(handler http.Handler, mapper api.RequestContextMapper, a
 				return
 			}
 
+			// authorization header is not required anymore in case of a successful authorization.
```

Comment:
also, update function godoc to describe the Authorization header being removed on success